### PR TITLE
Use CLI args for 'nda-transform/count_assoc_files.py' 

### PR DESCRIPTION
**Pull request by dheshanm**
Infers `nda_submissions_path`, `source_path` from the passed in `-id` and `-collection` CLI arguments

### CODE REVIEW

File: nda-transform/count_assoc_files.py [hunk 1]
```diff
@@ -1,81 +1,138 @@
...
-                        print(f"File does not exist: {data_file_path}")
-                    else:
-                        count+=1
-
-                        if data_file_path not in unique_file:
-                            unique_file[data_file_path]=1
+                check_file = True
+                break
+
+        if not check_file:
```
Comment (tashrifbillah) [inline nda-transform/count_assoc_files.py:48]: Dheshan, please remove this block and reinstate mine. it is too verbose as most files do not have them.

File: nda-transform/count_assoc_files.py [hunk 2]
```diff
@@ -1,81 +1,138 @@
...
-print('Duplicates', duplicate)
 
+# Usage:
+# at AWS workspace:
+# __file__  -id 65021 -collection 3705
+# at ERIS cluster:
+# __file__ -id 0 -collection 0
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Count unique and duplicate files")
+    parser.add_argument("-id", type=str, help="submission id", default="0")
```
Comment (tashrifbillah) [inline nda-transform/count_assoc_files.py:87]: Change to `('-i', '--id',` . Remember, the convention is that you use `--` for long argument and `-` for single letter argument.

File: nda-transform/count_assoc_files.py (hunk 1 above)
Comment (dheshanm) [reply nda-transform/count_assoc_files.py:48]: You want me to replace my block:

```python
        # check a file only if at least one file_columns exist
        check_file = False
        for c in file_columns:
            if c in df.columns:
                check_file = True
                break
```

with this block from yours:
```python
    # check a file only if at least one file_columns exist
    exist=0
    for c in file_columns:
        if c in df.columns:
            exist=1
            break
    if not exist:
        continue
```

File: nda-transform/count_assoc_files.py [hunk 3]
```diff
@@ -1,81 +1,138 @@
...
+    parser.add_argument(
+        "-collection",
```
Comment (tashrifbillah) [inline nda-transform/count_assoc_files.py:89]: Same here. Change to `('-c', '--collection',`.

File: nda-transform/count_assoc_files.py (hunk 1 above)
Comment (tashrifbillah) [reply nda-transform/count_assoc_files.py:48]: No. Remove lines 48-52.

Comment (dheshanm) [reply nda-transform/count_assoc_files.py:48]: Should I then also remove L31:

```python
print(f"Checking {file}", end="")
```

Comment (tashrifbillah) [reply nda-transform/count_assoc_files.py:48]: Why is this `end=""` necessary? Why was my message not sufficient?

File: nda-transform/count_assoc_files.py [hunk 4]
```diff
@@ -97,8 +97,8 @@ def count_files(
     submission_id = args.id
     collection_id = args.collection
 
-    print(f"Submission id: {submission_id}")
-    print(f"Collection id: {collection_id}")
+    print(f"submission id: {submission_id}")
+    print(f"collection id: {collection_id}")
```
Comment (tashrifbillah) [inline nda-transform/count_assoc_files.py:101]: No that is now what I said. Please emend the help message with upper case. Upper case is more appropriate.

**Comment by dheshanm** [comment]
Hi Tashrif, I have resolved all outstanding suggestion.

File: nda-transform/count_assoc_files.py [hunk 5]
```diff
@@ -1,81 +1,122 @@
...
+                        if not data_file_path.exists():
+                            print(f"File does not exist: {data_file_path}")
                         else:
-                            # count duplicates
-                            manifest=file.name
-                            if manifest in duplicate:
-                                duplicate[manifest]+=1
+                            files_count += 1
+                            if data_file_path not in unique_files:
+                                unique_files.add(data_file_path)
```
Comment (tashrifbillah) [inline nda-transform/count_assoc_files.py:63]: Dheshan, I believe it is the fastest in Python to build a set by using `dict[a]=''` method. How well do you know about `Set()` method?

File: nda-transform/count_assoc_files.py [hunk 6]
```diff
@@ -1,81 +1,122 @@
...
+
+        # check a file only if at least one file_columns exist
+        check_file = False
         for c in file_columns:
-        
             if c in df.columns:
-                if not pd.isna(row[c]):
-                    data_file = row[c]
+                check_file = True
+                break
```
Comment (tashrifbillah) [inline nda-transform/count_assoc_files.py:46]: Dheshan, I do not think your logic is right. Let's talk more about collaboration through GitHub. You should really change the things that are requested. If you major-revise the program, that makes it hard for your collaborator to follow.

**Comment by tashrifbillah** [comment]
Thank you Dheshan for this work.

**Pull request merged**